olive/passes/onnx/optimum_conversion.py

```diff
@@ -48,18 +48,30 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
                 default_value=None,
                 description="Extra arguments to pass to the `optimum.exporters.onnx.main_export` function.",
             ),
+            "custom_model": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Whether the model is a custom model which can not load from transformers directly",
+            ),
         }
 
     def validate_search_point(
         self, search_point: Dict[str, Any], accelerator_spec: AcceleratorSpec, with_fixed_value: bool = False
     ) -> bool:
+        from optimum import version as optimum_version
+        from packaging import version
+
         if with_fixed_value:
             search_point = self.config_at_search_point(search_point or {})
 
         if search_point.get("fp16") and search_point.get("device") != "cuda":
             logger.info("OptimumConversion: fp16 is set to True, but device is not set to cuda.")
             return False
 
+        if search_point.get("custom_model") and version.parse(optimum_version.__version__) < version.parse("1.17.0"):
+            logger.info("Custom model export is only supported in Optimum version 1.17.0 or later.")
+            return False
+
```

Comment:
Maybe move this part of filter to _run_for_config?  Just ignore custome_model when optimum version is not expected. Then logger.warning to tell users.